### PR TITLE
Added tests for te11.3 and related changes

### PR DESCRIPTION
**Pull request by Wanmathy**

### CODE REVIEW

**Comment by Wanmathy** [comment]
Adding @vosipchu,

Hi Viktor,

Could you please take a look at the PR.

Thanks,
Wanmathy

**Comment by guoshiuan** [comment]
Both sub-tests failed with 100% traffic loss:

* TestBackupNHGAction/testbackupDecap
    [third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go:215](https://cs.corp.google.com/piper///depot/google3/third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go?l=215&ws=guoshiuan/8880&snapshot=26): Name: testbackupDecap
    [third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go:216](https://cs.corp.google.com/piper///depot/google3/third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go?l=216&ws=guoshiuan/8880&snapshot=26): Description: Usecase3 with 2 NHOP Groups - Backup Pointing to Decap
    [third_party/openconfig/featureprofiles/internal/gribi/gribi.go:163](https://cs.corp.google.com/piper///depot/google3/third_party/openconfig/featureprofiles/internal/gribi/gribi.go?l=163&ws=guoshiuan/8880&snapshot=26): Trying to be a master with increasing the election id by one on dut.
    [third_party/openconfig/featureprofiles/internal/gribi/gribi.go:163](https://cs.corp.google.com/piper///depot/google3/third_party/openconfig/featureprofiles/internal/gribi/gribi.go?l=163&ws=guoshiuan/8880&snapshot=26): Learn GRIBI Election ID from dut.
    [third_party/openconfig/featureprofiles/internal/gribi/gribi.go:163](https://cs.corp.google.com/piper///depot/google3/third_party/openconfig/featureprofiles/internal/gribi/gribi.go?l=163&ws=guoshiuan/8880&snapshot=26): Setting GRIBI Election ID for dut to low=2, high=0
    [third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go:235](https://cs.corp.google.com/piper///depot/google3/third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go?l=235&ws=guoshiuan/8880&snapshot=26): Adding NH 1 with ATE port-2 via gRIBI
    [third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go:237](https://cs.corp.google.com/piper///depot/google3/third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go?l=237&ws=guoshiuan/8880&snapshot=26): Adding NH 2 as decap and NHGs 1, 2 via gRIBI
    [third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go:241](https://cs.corp.google.com/piper///depot/google3/third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go?l=241&ws=guoshiuan/8880&snapshot=26): Adding an IPv4Entry for 203.0.113.1 with primary atePort2, backup as Decap via gRIBI
    [third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go:244](https://cs.corp.google.com/piper///depot/google3/third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go?l=244&ws=guoshiuan/8880&snapshot=26): Create flow with dst 203.0.113.1
    [third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go:247](https://cs.corp.google.com/piper///depot/google3/third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go?l=247&ws=guoshiuan/8880&snapshot=26): Validate traffic passes
    [third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go:335](https://cs.corp.google.com/piper///depot/google3/third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go?l=335&ws=guoshiuan/8880&snapshot=26): 
        *** Starting traffic on ixia03-den86.corp.google.com...
        
        
    [third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go:337](https://cs.corp.google.com/piper///depot/google3/third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go?l=337&ws=guoshiuan/8880&snapshot=26): 
        *** Stopping traffic on ixia03-den86.corp.google.com...
        
        
    [third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go:250](https://cs.corp.google.com/piper///depot/google3/third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go?l=250&ws=guoshiuan/8880&snapshot=26): Shutdown Port2
    [third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go:252](https://cs.corp.google.com/piper///depot/google3/third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go?l=252&ws=guoshiuan/8880&snapshot=26): 
        *** Setting port state on ixia03-den86.corp.google.com...
        
        
    [third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go:256](https://cs.corp.google.com/piper///depot/google3/third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go?l=256&ws=guoshiuan/8880&snapshot=26): Capture port2 status if down
    [third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go:263](https://cs.corp.google.com/piper///depot/google3/third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go?l=263&ws=guoshiuan/8880&snapshot=26): Validate traffic passes through port3
    [third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go:335](https://cs.corp.google.com/piper///depot/google3/third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go?l=335&ws=guoshiuan/8880&snapshot=26): 
        *** Starting traffic on ixia03-den86.corp.google.com...
        
        
    [third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go:337](https://cs.corp.google.com/piper///depot/google3/third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go?l=337&ws=guoshiuan/8880&snapshot=26): 
        *** Stopping traffic on ixia03-den86.corp.google.com...
        
        
    [third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go:340](https://cs.corp.google.com/piper///depot/google3/third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go?l=340&ws=guoshiuan/8880&snapshot=26): LossPct for flow BackupFlow: got 100, want 0
    [third_party/go/gc/src/runtime/panic.go:522](https://cs.corp.google.com/piper///depot/google3/third_party/go/gc/src/runtime/panic.go?l=522&ws=guoshiuan/8880&snapshot=26): 
        *** Setting port state on ixia03-den86.corp.google.com...

* TestBackupNHGAction/testDecapEncap
    [third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go:215](https://cs.corp.google.com/piper///depot/google3/third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go?l=215&ws=guoshiuan/8880&snapshot=26): Name: testDecapEncap
    [third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go:216](https://cs.corp.google.com/piper///depot/google3/third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go?l=216&ws=guoshiuan/8880&snapshot=26): Description: Usecase3 with 2 NHOP Groups - Primary DecapEncap, Backup Pointing to Decap
    [third_party/openconfig/featureprofiles/internal/gribi/gribi.go:163](https://cs.corp.google.com/piper///depot/google3/third_party/openconfig/featureprofiles/internal/gribi/gribi.go?l=163&ws=guoshiuan/8880&snapshot=26): Trying to be a master with increasing the election id by one on dut.
    [third_party/openconfig/featureprofiles/internal/gribi/gribi.go:163](https://cs.corp.google.com/piper///depot/google3/third_party/openconfig/featureprofiles/internal/gribi/gribi.go?l=163&ws=guoshiuan/8880&snapshot=26): Learn GRIBI Election ID from dut.
    [third_party/openconfig/featureprofiles/internal/gribi/gribi.go:163](https://cs.corp.google.com/piper///depot/google3/third_party/openconfig/featureprofiles/internal/gribi/gribi.go?l=163&ws=guoshiuan/8880&snapshot=26): Setting GRIBI Election ID for dut to low=3, high=0
    [third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go:270](https://cs.corp.google.com/piper///depot/google3/third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go?l=270&ws=guoshiuan/8880&snapshot=26): Adding NH 3 with ATE port-2 via gRIBI
    [third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go:272](https://cs.corp.google.com/piper///depot/google3/third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go?l=272&ws=guoshiuan/8880&snapshot=26): Adding NHG 3 via gRIBI
    [third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go:274](https://cs.corp.google.com/piper///depot/google3/third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go?l=274&ws=guoshiuan/8880&snapshot=26): Adding an IPv4Entry for 203.0.113.70 pointing to atePort2 via gRIBI
    [third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go:277](https://cs.corp.google.com/piper///depot/google3/third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go?l=277&ws=guoshiuan/8880&snapshot=26): Adding NH 2 as decap and NHG 2 via gRIBI
    [third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go:281](https://cs.corp.google.com/piper///depot/google3/third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go?l=281&ws=guoshiuan/8880&snapshot=26): Adding NH 1 as DecapEncap and NHG 1 via gRIBI
    [third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go:284](https://cs.corp.google.com/piper///depot/google3/third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go?l=284&ws=guoshiuan/8880&snapshot=26): Adding an IPv4Entry for 203.0.113.1 with primary DecapEncap & backup decap via gRIBI
    [third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go:288](https://cs.corp.google.com/piper///depot/google3/third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go?l=288&ws=guoshiuan/8880&snapshot=26): Capture port2 status if Up
    [third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go:295](https://cs.corp.google.com/piper///depot/google3/third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go?l=295&ws=guoshiuan/8880&snapshot=26): Create base flow with dst 203.0.113.1
    [third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go:298](https://cs.corp.google.com/piper///depot/google3/third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go?l=298&ws=guoshiuan/8880&snapshot=26): Validate traffic passes through port2
    [third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go:335](https://cs.corp.google.com/piper///depot/google3/third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go?l=335&ws=guoshiuan/8880&snapshot=26): 
        *** Starting traffic on ixia03-den86.corp.google.com...
        
        
    [third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go:337](https://cs.corp.google.com/piper///depot/google3/third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go?l=337&ws=guoshiuan/8880&snapshot=26): 
        *** Stopping traffic on ixia03-den86.corp.google.com...
        
        
    [third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go:301](https://cs.corp.google.com/piper///depot/google3/third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go?l=301&ws=guoshiuan/8880&snapshot=26): Shutdown Port2
    [third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go:303](https://cs.corp.google.com/piper///depot/google3/third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go?l=303&ws=guoshiuan/8880&snapshot=26): 
        *** Setting port state on ixia03-den86.corp.google.com...
        
        
    [third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go:306](https://cs.corp.google.com/piper///depot/google3/third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go?l=306&ws=guoshiuan/8880&snapshot=26): Capture port2 status if down
    [third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go:313](https://cs.corp.google.com/piper///depot/google3/third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go?l=313&ws=guoshiuan/8880&snapshot=26): Validate traffic passes through port3
    [third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go:335](https://cs.corp.google.com/piper///depot/google3/third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go?l=335&ws=guoshiuan/8880&snapshot=26): 
        *** Starting traffic on ixia03-den86.corp.google.com...
        
        
    [third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go:337](https://cs.corp.google.com/piper///depot/google3/third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go?l=337&ws=guoshiuan/8880&snapshot=26): 
        *** Stopping traffic on ixia03-den86.corp.google.com...
        
        
    [third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go:340](https://cs.corp.google.com/piper///depot/google3/third_party/openconfig/featureprofiles/feature/experimental/backup_nhg_action/ate_tests/backup_nhg_action_test.go?l=340&ws=guoshiuan/8880&snapshot=26): LossPct for flow BackupFlow: got 100, want 0
    [third_party/go/gc/src/runtime/panic.go:522](https://cs.corp.google.com/piper///depot/google3/third_party/go/gc/src/runtime/panic.go?l=522&ws=guoshiuan/8880&snapshot=26): 
        *** Setting port state on ixia03-den86.corp.google.com...

**Comment by Wanmathy** [comment]
Could you please check if arp is resolved on port3.

Thanks.

**Comment by Wanmathy** [comment]
Checked with Vibov @vbhan-cisco on this. This is a known issue due to default route in static route configuration.

File: feature/experimental/gribi/ate_tests/backup_nhg_action/backup_nhg_action_test.go [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+	gnmi.Update(t, dut, d.NetworkInstance(*deviations.DefaultNetworkInstance).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_STATIC, *deviations.StaticProtocolName).Config(), static)
+}
+
+// configureNetworkInstance configures vrf VRF-1 and adds the vrf to port1.
+func configureNetworkInstance(t *testing.T, dut *ondatra.DUTDevice) {
+	c := &oc.Root{}
+	ni := c.GetOrCreateNetworkInstance(vrfName)
+	ni.Description = ygot.String("Non Default routing instance created for testing")
+	ni.Type = oc.NetworkInstanceTypes_NETWORK_INSTANCE_TYPE_L3VRF
+	ni.Enabled = ygot.Bool(true)
```
Comment (prinikasn) [inline feature/experimental/gribi/ate_tests/backup_nhg_action/backup_nhg_action_test.go:162]: ni.Enabled undefined NetworkInstance has no field or method Enabled and EnableAddressFamilies

File: https://raw.githubusercontent.com/openconfig/ondatra/main/gnmi/oc/structs-0.go

// NetworkInstance represents the /openconfig-network-instance/network-instances/network-instance YANG schema element.
type NetworkInstance struct {
	Afts                    *NetworkInstance_Afts                                                    `path:"afts" module:"openconfig-network-instance"`
	ConnectionPoint         map[string]*NetworkInstance_ConnectionPoint                              `path:"connection-points/connection-point" module:"openconfig-network-instance/openconfig-network-instance"`
	Description             *string                                                                  `path:"state/description" module:"openconfig-network-instance/openconfig-network-instance" shadow-path:"config/description" shadow-module:"openconfig-network-instance/openconfig-network-instance"`
	Encapsulation           *NetworkInstance_Encapsulation                                           `path:"encapsulation" module:"openconfig-network-instance"`
	Evpn                    *NetworkInstance_Evpn                                                    `path:"evpn" module:"openconfig-network-instance"`
	FallbackNetworkInstance *string                                                                  `path:"state/fallback-network-instance" module:"openconfig-network-instance/openconfig-network-instance" shadow-path:"config/fallback-network-instance" shadow-module:"openconfig-network-instance/openconfig-network-instance"`
	Fdb                     *NetworkInstance_Fdb                                                     `path:"fdb" module:"openconfig-network-instance"`
	InterInstancePolicies   *NetworkInstance_InterInstancePolicies                                   `path:"inter-instance-policies" module:"openconfig-network-instance"`
	Interface               map[string]*NetworkInstance_Interface                                    `path:"interfaces/interface" module:"openconfig-network-instance/openconfig-network-instance"`
	Mpls                    *NetworkInstance_Mpls                                                    `path:"mpls" module:"openconfig-network-instance"`
	Name                    *string                                                                  `path:"state/name|name" module:"openconfig-network-instance/openconfig-network-instance|openconfig-network-instance" shadow-path:"config/name|name" shadow-module:"openconfig-network-instance/openconfig-network-instance|openconfig-network-instance"`
	PolicyForwarding        *NetworkInstance_PolicyForwarding                                        `path:"policy-forwarding" module:"openconfig-network-instance"`
	Protocol                map[NetworkInstance_Protocol_Key]*NetworkInstance_Protocol               `path:"protocols/protocol" module:"openconfig-network-instance/openconfig-network-instance"`
	RouteDistinguisher      *string                                                                  `path:"state/route-distinguisher" module:"openconfig-network-instance/openconfig-network-instance" shadow-path:"config/route-distinguisher" shadow-module:"openconfig-network-instance/openconfig-network-instance"`
	RouteLimit              map[E_Types_ADDRESS_FAMILY]*NetworkInstance_RouteLimit                   `path:"route-limits/route-limit" module:"openconfig-network-instance/openconfig-network-instance"`
	RouterId                *string                                                                  `path:"state/router-id" module:"openconfig-network-instance/openconfig-network-instance" shadow-path:"config/router-id" shadow-module:"openconfig-network-instance/openconfig-network-instance"`
	SegmentRouting          *NetworkInstance_SegmentRouting                                          `path:"segment-routing" module:"openconfig-network-instance"`
	Table                   map[NetworkInstance_Table_Key]*NetworkInstance_Table                     `path:"tables/table" module:"openconfig-network-instance/openconfig-network-instance"`
	TableConnection         map[NetworkInstance_TableConnection_Key]*NetworkInstance_TableConnection `path:"table-connections/table-connection" module:"openconfig-network-instance/openconfig-network-instance"`
	Type                    E_NetworkInstanceTypes_NETWORK_INSTANCE_TYPE                             `path:"state/type" module:"openconfig-network-instance/openconfig-network-instance" shadow-path:"config/type" shadow-module:"openconfig-network-instance/openconfig-network-instance"`
	Vlan                    map[uint16]*NetworkInstance_Vlan                                         `path:"vlans/vlan" module:"openconfig-network-instance/openconfig-network-instance"`
}

File: feature/experimental/gribi/ate_tests/backup_nhg_action/backup_nhg_action_test.go [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+	ni.EnabledAddressFamilies = []oc.E_Types_ADDRESS_FAMILY{oc.Types_ADDRESS_FAMILY_IPV4, oc.Types_ADDRESS_FAMILY_IPV6}
```
Comment (prinikasn) [inline feature/experimental/gribi/ate_tests/backup_nhg_action/backup_nhg_action_test.go:163]: Same as above.

File: feature/experimental/gribi/ate_tests/backup_nhg_action/backup_nhg_action_test.go (hunk 1 above)
Comment (prinikasn) [reply feature/experimental/gribi/ate_tests/backup_nhg_action/backup_nhg_action_test.go:162]: Ref: Looks like this field was removed in PR https://github.com/openconfig/public/pull/749

**Review by prinikasn — COMMENTED** [review]
/gcbrun

**Review by prinikasn — COMMENTED** [review]
/gcbrun

File: feature/experimental/gribi/ate_tests/backup_nhg_action/backup_nhg_action_test.go [hunk 3]
```diff
@@ -0,0 +1,343 @@
...
+	i3.IPv4().
+		WithAddress(atePort3.IPv4CIDR()).
+		WithDefaultGateway(dutPort3.IPv4)
+	i3.IPv6().
+		WithAddress(atePort3.IPv6CIDR()).
+		WithDefaultGateway(dutPort3.IPv6)
+	return top
+}
+
+// configureDUT configures port1, port2, port3 and port4 on the DUT.
```
Comment (thesrinath) [inline feature/experimental/gribi/ate_tests/backup_nhg_action/backup_nhg_action_test.go:131]: There's no port4

File: feature/experimental/gribi/ate_tests/backup_nhg_action/backup_nhg_action_test.go [hunk 4]
```diff
@@ -0,0 +1,343 @@
...
+	top := configureATE(t, ate)
+	top.Push(t).StartProtocols(t)
+
+	test := []struct {
+		name string
+		desc string
+		fn   func(ctx context.Context, t *testing.T, args *testArgs)
+	}{
+		{
+			name: "testbackupDecap",
```
Comment (thesrinath) [inline feature/experimental/gribi/ate_tests/backup_nhg_action/backup_nhg_action_test.go:190]: testBackupDecap?

File: feature/experimental/gribi/ate_tests/backup_nhg_action/backup_nhg_action_test.go [hunk 5]
```diff
@@ -0,0 +1,343 @@
...
+			desc: "Usecase3 with 2 NHOP Groups - Backup Pointing to Decap",
+			fn:   testbackupDecap,
```
Comment (thesrinath) [inline feature/experimental/gribi/ate_tests/backup_nhg_action/backup_nhg_action_test.go:192]: ditto

File: feature/experimental/gribi/ate_tests/backup_nhg_action/backup_nhg_action_test.go [hunk 6]
```diff
@@ -0,0 +1,343 @@
...
+		},
+		{
+			name: "testDecapEncap",
+			desc: "Usecase3 with 2 NHOP Groups - Primary DecapEncap, Backup Pointing to Decap",
+			fn:   testDecapEncap,
+		},
+	}
+	// Configure the gRIBI client client
```
Comment (thesrinath) [inline feature/experimental/gribi/ate_tests/backup_nhg_action/backup_nhg_action_test.go:200]: remove 2nd client

File: feature/experimental/gribi/ate_tests/backup_nhg_action/backup_nhg_action_test.go [hunk 7]
```diff
@@ -0,0 +1,343 @@
...
+				ate:    ate,
+				top:    top,
+			}
+			tt.fn(ctx, t, tcArgs)
+		})
+	}
+}
+
+// TE11.3 - case 1: next-hop viability triggers decap in backup NHG.
+func testbackupDecap(ctx context.Context, t *testing.T, args *testArgs) {
```
Comment (thesrinath) [inline feature/experimental/gribi/ate_tests/backup_nhg_action/backup_nhg_action_test.go:231]: testBackupDecap

File: feature/experimental/gribi/ate_tests/backup_nhg_action/backup_nhg_action_test.go [hunk 8]
```diff
@@ -0,0 +1,343 @@
...
+
+	t.Logf("Adding NH %d with ATE port-2 via gRIBI", NH1ID)
```
Comment (thesrinath) [inline feature/experimental/gribi/ate_tests/backup_nhg_action/backup_nhg_action_test.go:233]: atePort2 to be consistent with line 239

File: feature/experimental/gribi/ate_tests/backup_nhg_action/backup_nhg_action_test.go [hunk 9]
```diff
@@ -0,0 +1,343 @@
...
+	innerIpv4Header.WithSrcAddress(innersrcPfx)
+	innerIpv4Header.DstAddressRange().WithMin(innerdstPfx).WithCount(1)
+	flow := ate.Traffic().NewFlow(name).
+		WithSrcEndpoints(srcEndPoint).
+		WithDstEndpoints(dstEndPoint).
+		WithHeaders(ondatra.NewEthernetHeader(), hdr, innerIpv4Header).WithFrameSize(300)
+	return flow
+}
+
+// validateTrafficFlows verifies that the flow on ATE good flow delivers traffic, bad flow doesnt.
```
Comment (thesrinath) [inline feature/experimental/gribi/ate_tests/backup_nhg_action/backup_nhg_action_test.go:331]: please check grammar

**Review by thesrinath — APPROVED** [review]

**Pull request merged**